File: src/frontend/src/ui/connectors/categories/ConnectorCategory.ts
/**
 * Descriptor for a connector category.
 *
 * @param name - The category name.
 * @param description - An optional description.
 */
export class ConnectorCategory {
    public readonly name: string;
    public readonly description: string;

    protected constructor(name: string, description: string) {
        this.name = name;
        this.description = description;
    }
}

File: src/common/py/data/entities/connector/categories/connector_category.py
from dataclasses import dataclass

ConnectorCategoryID = str


@dataclass(frozen=True, kw_only=True)
class ConnectorCategory:
    """
    Descriptor for a connector category.

    Attributes:
        name: The category name.
        description: An optional description.
    """

    name: str
    description: str
